app/Http/Controllers/BeatmapsController.php

```diff
@@ -64,20 +88,24 @@ public function scores($id)
                 ->where('beatmap_id', $beatmap->getKey())
                 ->with(['beatmap', 'user.country', 'user.userProfileCustomization'])
                 ->withMods($mods)
-                ->withType($type, compact('user'));
+                ->withType($type, ['user' => $currentUser]);
+
+            if ($currentUser !== null && $userId === null) {
+                $userScore = (clone $query)->where('user_id', $currentUser->user_id)->first();
+            }
 
-            if ($user !== null) {
-                $score = (clone $query)->where('user_id', $user->user_id)->first();
+            if ($userId !== null) {
+                $query->where('user_id', $userId);
             }
 
             $results = [
                 'scores' => json_collection($query->visibleUsers()->forListing(), 'Score', ['beatmap', 'user', 'user.country', 'user.cover']),
```

Comment:
`forListing` needs adjustment to not deduplicate user scores when requesting scores of specific user.

Reply:
for some reason I thought `forListing` did the sorting but it doesn't 🤔